tsgo panics when emitting a `do` statement with no `while`

## Stack trace

```
panic: runtime error: slice bounds out of range [7:2] [recovered, repanicked]

goroutine 93 [running]:
sync.(*WaitGroup).Go.func1.1()
        sync/waitgroup.go:251 +0x48
panic({0x1054d7d80?, 0x73bed7608018?})
        runtime/panic.go:860 +0x12c
github.com/microsoft/typescript-go/internal/scanner.SkipTriviaEx({0x73bed5e9c400, 0x2}, 0x73bed5e9c400?, 0x0?)
        github.com/microsoft/typescript-go/internal/scanner/scanner.go:2281 +0x46c
github.com/microsoft/typescript-go/internal/scanner.SkipTrivia(...)
        github.com/microsoft/typescript-go/internal/scanner/scanner.go:2266
github.com/microsoft/typescript-go/internal/printer.(*Printer).emitCommentsBeforeToken(0x73bed676e700, 0x6d40?, 0x7, 0x73bed5edc000, 0x0)
        github.com/microsoft/typescript-go/internal/printer/printer.go:5289 +0x58
github.com/microsoft/typescript-go/internal/printer.(*Printer).enterToken(0x73bed676e700, 0x14, 0x7?, 0x73bed5edc000, 0x0)
        github.com/microsoft/typescript-go/internal/printer/printer.go:6117 +0x2c
github.com/microsoft/typescript-go/internal/printer.(*Printer).emitTokenEx(0x73bed676e700, 0x14, 0x73bed65a4fc0?, 0x3, 0x73bed5edc000, 0x2?)
        github.com/microsoft/typescript-go/internal/printer/printer.go:924 +0x34
github.com/microsoft/typescript-go/internal/printer.(*Printer).emitToken(...)
        github.com/microsoft/typescript-go/internal/printer/printer.go:920
github.com/microsoft/typescript-go/internal/printer.(*Printer).emitWhileClause(0x73bed676e700, 0x73bed5edc000, 0x73bed5eba168, 0x73bed5eba168?)
        github.com/microsoft/typescript-go/internal/printer/printer.go:3452 +0x84
github.com/microsoft/typescript-go/internal/printer.(*Printer).emitDoStatement(0x73bed676e700, 0x73bed5edc000)
        github.com/microsoft/typescript-go/internal/printer/printer.go:3467 +0xc8
github.com/microsoft/typescript-go/internal/printer.(*Printer).emitStatement(0xffffffffffffffff?, 0x73bed7405a68?)
        github.com/microsoft/typescript-go/internal/printer/printer.go:4126 +0xe0
github.com/microsoft/typescript-go/internal/printer.(*Printer).emitListItems(0x73bed676e700, 0x10564b660, 0x73bed7aff508, {0x73bed7703568, 0x1, 0x73bed7aff508?}, 0x1, 0x0, {0x1?, 0xd676e700?})
        github.com/microsoft/typescript-go/internal/printer/printer.go:4928 +0x4d4
github.com/microsoft/typescript-go/internal/printer.(*Printer).emitListRange(0x73bed676e700, 0x10564b660, 0x73bed7aff508, 0x73bed6411ac0, 0x1, 0x73bed7442028?, 0x73bed7405b68?)
        github.com/microsoft/typescript-go/internal/printer/printer.go:4727 +0x310
github.com/microsoft/typescript-go/internal/printer.(*Printer).emitSourceFile(0x73bed676e700, 0x73bed7aff508)
        github.com/microsoft/typescript-go/internal/printer/printer.go:4632 +0x21c
github.com/microsoft/typescript-go/internal/printer.(*Printer).Write(0x73bed676e700, 0x73bed7aff508, 0x73bed7aff508, {0x1056720b0, 0x73bed65a4fc0}, 0x105656f98?)
        github.com/microsoft/typescript-go/internal/printer/printer.go:5190 +0xc40
github.com/microsoft/typescript-go/internal/compiler.(*emitter).printSourceFile(0x73bed6377960, {0x73bed5de83c0, 0x2b}, {0x0, 0x0}, 0x73bed7aff508, 0x73bed676e700, 0x0)
        github.com/microsoft/typescript-go/internal/compiler/emitter.go:278 +0x194
github.com/microsoft/typescript-go/internal/compiler.(*emitter).emitJSFile(0x73bed6377960, 0x73bed5ede008, {0x73bed5de83c0, 0x2b}, {0x0, 0x0})
        github.com/microsoft/typescript-go/internal/compiler/emitter.go:210 +0x2e8
github.com/microsoft/typescript-go/internal/compiler.(*emitter).emit(0x73bed6377960)
        github.com/microsoft/typescript-go/internal/compiler/emitter.go:49 +0xec
github.com/microsoft/typescript-go/internal/compiler.(*Program).Emit.func2()
        github.com/microsoft/typescript-go/internal/compiler/program.go:1649 +0x194
github.com/microsoft/typescript-go/internal/core.(*parallelWorkGroup).Queue.func1()
        github.com/microsoft/typescript-go/internal/core/workgroup.go:40 +0x24
sync.(*WaitGroup).Go.func1()
        sync/waitgroup.go:258 +0x48
created by sync.(*WaitGroup).Go in goroutine 1
        sync/waitgroup.go:238 +0x70
```

## Steps to reproduce

```sh
echo 'do' > bug.ts
tsgo bug.ts
```

--- a/internal/scanner/scanner.go
+++ b/internal/scanner/scanner.go
@@ -2278,6 +2278,9 @@ func SkipTriviaEx(text string, pos int, options *SkipTriviaOptions) int {
 	canConsumeStar := false
 	// Keep in sync with couldStartTrivia
 	for {
+		if pos >= textLen {
+			return pos
+		}
 		ch, size := utf8.DecodeRuneInString(text[pos:])
 		switch ch {
 		case '\r':

--- a/testdata/baselines/reference/compiler/emitIncompleteDoStatement.errors.txt
+++ b/testdata/baselines/reference/compiler/emitIncompleteDoStatement.errors.txt
@@ -1,0 +1,7 @@
+a.ts(1,3): error TS1109: Expression expected.
+
+
+==== a.ts (1 errors) ====
+    do
+      
+!!! error TS1109: Expression expected.

--- a/testdata/baselines/reference/compiler/emitIncompleteDoStatement.js
+++ b/testdata/baselines/reference/compiler/emitIncompleteDoStatement.js
@@ -1,0 +1,10 @@
+//// [tests/cases/compiler/emitIncompleteDoStatement.ts] ////
+
+//// [a.ts]
+do
+
+//// [a.js]
+"use strict";
+do
+    ;
+while ();

--- a/testdata/baselines/reference/compiler/emitIncompleteDoStatement.symbols
+++ b/testdata/baselines/reference/compiler/emitIncompleteDoStatement.symbols
@@ -1,0 +1,5 @@
+//// [tests/cases/compiler/emitIncompleteDoStatement.ts] ////
+
+=== a.ts ===
+
+do

--- a/testdata/baselines/reference/compiler/emitIncompleteDoStatement.types
+++ b/testdata/baselines/reference/compiler/emitIncompleteDoStatement.types
@@ -1,0 +1,7 @@
+//// [tests/cases/compiler/emitIncompleteDoStatement.ts] ////
+
+=== a.ts ===
+do
+> : any
+> : any
+

--- a/testdata/tests/cases/compiler/emitIncompleteDoStatement.ts
+++ b/testdata/tests/cases/compiler/emitIncompleteDoStatement.ts
@@ -1,0 +1,2 @@
+// @filename: a.ts
+do

